File: Channel.rb
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# Channel.rb

require "./Logger.rb"
require "./RssInfo.rb"
require "./Item.rb"

class Channel
    attr_reader :hasTitle, :hasLink, :hasDescription, :items
    
    public
        def initialize()
            @hasTitle = false
            @hasLink = false
            @hasDescription = false
            @items = Array.new()
        end
        
        def foundTitle()
            if not @hasTitle
                @hasTitle = true
            else
                Logger.instance.writeError( "Error: A channel can only have one title." )
                RssInfo.instance.errorsFound += 1
            end
        end
        
        def foundLink()
            if not @hasLink
                @hasLink = true
            else
                Logger.instance.writeError( "Error: A channel can only have one link." )
                RssInfo.instance.errorsFound += 1
            end
        end
        
        def foundDescription()
            if not @hasDescription
                @hasDescription = true
            else
                Logger.instance.writeError( "Error: A channel can only have one description." )
                RssInfo.instance.errorsFound += 1
            end
        end
        
        def foundItem()
            @items.push( Item.new() )
        end
end

File: Channel.py
# encoding: utf-8
# Rodrigo Fuentes Hernandez
# aka Blurred_9L
# Compiladores CUCEI 2013B
# Channel.py

import Logger
import RssInfo
import Item

class Channel( object ):

    def __init__( self ):
        self.hasTitle = False
        self.hasLink = False
        self.hasDescription = False
        self.items = list()
        
    def foundTitle( self, rssInfo, log ):
        if not self.hasTitle:
            self.hasTitle = True
        else:
            log.writeError( "Error: A channel can only have one title." )
            rssInfo.errorsFound += 1
            
    def foundLink( self, rssInfo, log ):
        if not self.hasLink:
            self.hasLink = True
        else:
            log.writeError( "Error: A channel can only have one link." )
            rssInfo.errorsFound += 1
            
    def foundDescription( self, rssInfo, log ):
        if not self.hasDescription:
            self.hasDescription = True
        else:
            log.writeError( "Error: A channel can only have one link." )
            rssInfo.errorsFound += 1
            
    def foundItem( self ):
        self.items.append( Item.Item() )
        
    def isCorrect( self ):
        return self.hasTitle and self.hasLink and self.hasDescription
